doc/pick_place/scripts/pick_place_tutorial.py

```diff
@@ -0,0 +1,331 @@
+
+	## Define the primitive and its dimensions. ##
+	collision_object_sizes[1] = (0.4, 0.2, 0.4) # Box size
+
+	## Define the pose of the table. ##
+	collision_objects[1].pose.position.x = 0
+	collision_objects[1].pose.position.y = 0.5
+	collision_objects[1].pose.position.z = 0.2
+
+	## Define the object that we will be manipulating ##
+	collision_objects_names[2] = "object"
+	collision_objects[2].header.frame_id = "panda_link0"
+
+	## Define the primitive and its dimensions. ##
+	collision_object_sizes[2] = (0.02, 0.02, 0.2) # Box size
+
+	## Define the pose of the object. ##
+	collision_objects[2].pose.position.x = 0.5
+	collision_objects[2].pose.position.y = 0
+	collision_objects[2].pose.position.z = 0.5
+	## - END_SUB_TUTORIAL - ##
+
+	## Add collision objects to scene ##
+	for (name, pose, size) in zip(collision_objects_names, collision_objects, collision_object_sizes):
+		planning_scene_interface.add_box(name=name, pose=pose, size=size)
+
+def main(arg):
+	"""Main function"""
+
+	## Initialize ros node ##
+	rospy.init_node("panda_arm_pick_place")
+
+	## initialize moveit_commander ##
+	moveit_commander.roscpp_initialize(arg)
+
+	## Connect to moveit services ##
+	rospy.loginfo(	  "Conneting moveit default moveit \'apply_planning_scene\' service.")
+	rospy.wait_for_service('apply_planning_scene')
+	try:
+		planning_scene_srv = rospy.ServiceProxy('apply_planning_scene', ApplyPlanningScene)
+		rospy.loginfo("Moveit \'apply_planning_scene\' service found!")
+	except rospy.ServiceException as e:
+		rospy.logerr(
+			  "Moveit \'apply_planning_scene\' service initialization failed: %s" % e)
+		shutdown_msg = "Shutting down %s node because %s service connection failed." % (rospy.get_name(), planning_scene_srv.resolved_name)
+		rospy.logerr(shutdown_msg)
+		sys.exit(0)
+
+	## Create robot commander ##
+	robot = moveit_commander.RobotCommander(
+		robot_description="robot_description", ns="/")
+	rospy.logdebug("Robot Groups: %s", robot.get_group_names())
+
+	## Create scene commanders ##
+	# Used to get information about the world and update the robot
+	# its understanding of the world.
+	move_group = robot.get_group("panda_arm")
+	planning_scene_interface = moveit_commander.PlanningSceneInterface(ns="/")
+
+	## Specify the planner we want to use ##
+	move_group.set_planner_id("TRRTkConfigDefault")
+
+	## Create a `DisplayTrajectory`_ ROS publisher to display the plan in RVIZ ##
+	display_trajectory_publisher = rospy.Publisher('/move_group/display_planned_path',
+												   DisplayTrajectory,
+												   queue_size=20)
+
+    ## Wait a bit for ROS to initialize the planning_scene_interface ##
+	rospy.sleep(1.0)
+
+	## Add collision objects ##
+	addCollisionObjects(planning_scene_interface)
+
+	## Wait a bit ##
+	rospy.sleep(1.0)
+
+	## Pick ##
+	pick(move_group)
+
+	## Wait a bit ##
+	rospy.sleep(1.0)
+
+	## Place ##
+	place(move_group)
+
+if __name__ == "__main__":
+    main(sys.argv[1:])
```

Comment:
Remove sys.argv unless you really need it. rosrun causes this value to be different than if you ran it directly with the interpreter.

Reply:
@mlautman Thanks for bringing that to my attention. Based on the [move_group python tutorial](https://ros-planning.github.io/moveit_tutorials/doc/move_group_python_interface/move_group_python_interface_tutorial.html) and the [moveit_commander.roscpp_initializer code](https://docs.ros.org/jade/api/moveit_commander/html/namespacemoveit__commander_1_1roscpp__initializer.html) I assumed the `argv` argument was needed in the roscpp_initializer.